### ISSUE TITLE
Foundation: All xy-grid cells are width 8.3333333333%

### ISSUE THREAD

**Issue by berarma** [scssphp/scssphp#75]
I'm compiling Foundation 6.6.1 using scssphp 1.0.6 (I've tried the master branch too) and the compilation goes well but the following HTML code results in 3 columns with width 8.3333333333%:

```
    <div class="grid-x">
        <div class="cell medium-4">
          Column 1
        </div>
        <div class="cell medium-4">
          Column 2
        </div>
        <div class="cell medium-4">
          Column 3
        </div>
    </div>
```

The generated CSS has these code:
```
@media print, screen and (min-width: 40em) {
  .grid-x > .medium-auto {
    flex: 1 1 0px;
    width: auto; }
  .grid-x > .medium-shrink, .grid-x > .medium-1, .grid-x > .medium-2, .grid-x > .medium-3, .grid-x > .medium-4, .grid-x > .medium-5, .grid-x > .medium-6, .grid-x > .medium-7, .grid-x > .medium-8, .grid-x > .medium-9, .grid-x > .medium-10, .grid-x > .medium-11, .grid-x > .medium-12 {
    flex: 0 0 auto; }
  .grid-x > .medium-shrink {
    width: auto; }
  .grid-x > .medium-1 {
    width: 8.3333333333%; }
  .grid-x > .medium-2 {
    width: 8.3333333333%; }
  .grid-x > .medium-3 {
    width: 8.3333333333%; }
  .grid-x > .medium-4 {
    width: 8.3333333333%; }
  .grid-x > .medium-5 {
    width: 8.3333333333%; }
  .grid-x > .medium-6 {
    width: 8.3333333333%; }
  .grid-x > .medium-7 {
    width: 8.3333333333%; }
  .grid-x > .medium-8 {
    width: 8.3333333333%; }
  .grid-x > .medium-9 {
    width: 8.3333333333%; }
  .grid-x > .medium-10 {
    width: 8.3333333333%; }
  .grid-x > .medium-11 {
    width: 8.3333333333%; }
  .grid-x > .medium-12 {
    width: 8.3333333333%; } }
```

The result is so strange I thought it could be that this version of Foundation isn't supported. Which versions do you use? Anyone has ever seen this problem?

Thanks.

**Comment by berarma** [comment]
I've tried compiling with the Ruby sass implementation and it works correctly. It seems to be a bug in ScssPHP.

**Comment by robocoder** [comment]
Looks like composer.json and tests refer to Foundation 6.5.

Maybe we're not picking up 6.6 because of their packagist changes? ie for 6.6, they've changed the vendor name (from zurb/foundation) and are in the process of transferring the packagist account (for foundation/foundation).

**Comment by Cerdic** [comment]
@robocoder however in the test suite we're just checking scssphp is able to compile Foundation, not checking for the final result.

I manually checked the compilation result on BootStrap 4 when including it in the testsuite and it was ok (ie equivalent when not exactly the same), but it seems hard to have an automated test on the full compilation as there is too many litteral difference between compiler results even when the results are equivalent.
But I did not checked for Foundation cause I'm not really knowing it.

**Comment by Cerdic** [comment]
@berarma my experience on debuging scssphp is that Foundation is way more complicated than Bootstrap in its architecture and it's fully impossible to debug on the complete compilation.

But if you can reach a minimal test case by extracting just what is needed to reproduce the bug there is good chance that we can fix it :)

**Comment by Cerdic** [comment]
looks like 1f7a5174 improved a bit the final result, but 4ebc5e14 broke everything on this part - working on revert that

**Comment by Cerdic** [comment]
a744407c fix the breaking from 4ebc5e1 and 7dc647de fix the percentage issue
But there is still an issue a bit further  with something like
```
    .small-margin-collapse > .1-1 {
      width: 8.3333333333%; }
    .small-margin-collapse > .1-2 {
      width: 16.6666666667%; }
    .small-margin-collapse > .1-3 {
      width: 25%; }
```
This is again linked to scope vars (here `@content` vs `@mixin`) and I was able to get a minimal test case:

```
$breakpoint-classes: (small, medium);
$grid-columns: 3;

@mixin breakpoint() {
  $bp: 1;
  @content;
}

.grid {
  // Output new widths to not include gutters
  @each $bp in $breakpoint-classes {
    @include breakpoint() {
      @for $i from 1 through $grid-columns {
        // Sizing (percentage)
        > .#{$bp}-#{$i} {
          width: percentage($i/$grid-columns);
        }
      }
    }
  }
}
```

which gives:
```
.grid > .1-1 {
  width: 33.3333333333%; }
.grid > .1-2 {
  width: 66.6666666667%; }
.grid > .1-3 {
  width: 100%; }
.grid > .1-1 {
  width: 33.3333333333%; }
.grid > .1-2 {
  width: 66.6666666667%; }
.grid > .1-3 {
  width: 100%; }
```

**Comment by Cerdic** [comment]
And this is finaly fixed!
